odk1-src/form-audit-log.rst

```diff
@@ -0,0 +1,146 @@
+.. spelling::
+
+  timestamp
+
+Logging enumerator behavior
+=============================
+
+Collect can log the behavior of enumerators as they navigate through a form. This log has many uses including discovering:
+
+- questions that take a long time to answer
+
+- how enumerators typically navigate through a form
+
+- enumerators who take a particularly long or short time to answer
+
+This information can inform form design and training.
+
+.. admonition:: Aggregate 1.5.0+ required
+
+  If a version of Aggregate lower than 1.5.0 is used, **audit files will not be saved on the server**.
+
+.. contents:: :depth: 1
+  :local:
+
+.. _enabling-audit-logging:
+
+Enabling audit logging
+-----------------------
+
+To enable logging for a form, add a row of :th:`type` :tc:`audit` and :th:`name` :tc:`audit` in an XLSForm:
+
+.. csv-table:: survey
+  :header: type, name, label
+
+  audit, audit, 
+
+A form may contain at most one row of :th:`type` :tc:`audit`.
+
+.. _viewing-audit-logs:
+
+Viewing audit logs
+-------------------
+
+Audit logs can be reviewed in Aggregate and downloaded for further analysis using Briefcase.
+
+In Aggregate 1.5.0+, audit logs can be viewed by clicking on the media icon in the :tc:`meta audit` column on the Submissions page:
+
+.. image:: /img/form-audit-log/audit-media-icon.png
+  :alt: The Aggregate submissions page with a form that has an audit log. The media icon in the meta audit column is circled.
+
+This displays a popup with the audit contents:
+
+.. image:: /img/form-audit-log/audit-example.png
+  :alt: An example audit log in Aggregate.
+
+If more sophisticated analysis of the logs is required, they can be downloaded along with their submissions using :ref:`Briefcase <pull-from-aggregate>`.
+
+.. _audit-log-structure:
+
+Log structure
+---------------
+
+If a form includes an audit, Collect will create an ``audit.csv`` file as the form is filled out. The ``audit.csv`` file has the following structure:
+
+.. csv-table:: audit.csv
+  :header: event, node, start, end
+
+  question, /data/name, 1523403169208, 1523403170733
+
+Values in the ``event`` column represent a particular user action such as opening a form, saving a form or displaying a question. Possible event types are described in the :ref:`audit-event-types` section.
+
+Values in the ``node`` column represent the node in the form that the event refers to, if applicable.
+
+Values in the ``start`` and ``end`` columns are timestamps represented as the number of milliseconds since midnight, January 1, 1970 UTC. This is known as epoch time and provides a standard way of representing date/time even across timezones. The :ref:`audit-timestamps` section contains more information about timestamps.
+
+.. _audit-event-types:
+
+Event types
+--------------
+
+The event column of the audit log can have the following values:
+
++-------------------+------------------------------------------------------------------+-------+----------------+
```

Comment:
I'm not so sure about the format you use for `start`` here in the timestamps column. Perhaps @adammichaelwood has thoughts on if this is correct.

Reply:
I agree its weird. Typically the double-back-tick style is for literal strings (equivalent to `<code>`).

I might use the `:th:` text role, since you are referring back to info in the *start* column of the audit log.

Reply:
I'm also curious why you used this markup style for the table. It's fine, but strikes me as harder to write than the csv table markup.

Reply:
It didn't feel like a CSV! Do you use it for arbitrary tables? I used https://ozh.github.io/ascii-tables/ so it wasn't a big deal.

Reply:
I see what you mean, but yes --- I use it for arbitrary tables because I find the markup format to just be easier to read, type, and edit in source.